### PR TITLE
Blaze: Support evergreen campaign creation

### LINKED ISSUE #13524: Networking: Update `CreateBlazeCampaign` and `BlazeForecastedImpressionsInput` structs with the new field `is_evergreen`

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #13524 

## Description

This PR updates `CreateBlazeCampaign` struct and `BlazeForecastedImpressionsInput` with a new field `isEvergreen` to support creating evergreen campaigns for Blaze.

For the campaign creation to succeed, a few more changes have been added:
- Updated the budget type for evergreen campaigns to be `daily` because we cannot calculate the total budget for an unlimited campaign.
- Updated the confirm payment screen to display the weekly budget for evergreen campaigns. This is also the amount to be charged weekly when the campaign starts. 

## Steps to reproduce

- Enable the feature flag `blazeEvergreenCampaigns` and build the app.
- Log in to a store eligible for Blaze.
- Start the campaign creation flow for a published product.
- Select the budget field and drag the budget slider. Confirm with proxyman that the impression requests send the correct `is_evergreen` value. Repeat the test when toggle the "Run it until I stop" switch to confirm it's still correct.
- Set the option back to "Run it until I stop" and proceed to create the campaign.
- User proxyman to confirm that the campaign creation request sends the correct `is_evergreen` campaign value.
- Open the newly created campaign and confirm that the campaign is indeed evergreen.

Optional: disable the `blazeEvergreenCampaigns` feature flag and confirm that both the above requests send the default `is_evergreen` as false.

## Screenshots

Evergreen | Non evergreen
--- | ---
 | 

---
- [x] I have considered adding unit tests for this change. If I decided not to add them, I have provided a brief explanation below (optional):
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
- [ ] This PR includes refactoring; smoke testing of the entire section is needed.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift [hunk 1]
```diff
@@ -232,6 +245,29 @@ private extension BlazeConfirmPaymentViewModel {
     }
 }
 
+private extension BlazeConfirmPaymentViewModel {
+    enum Localization {
+        static let totalWeeklyAmount = NSLocalizedString(
+            "blazeConfirmPaymentViewModel.totalWeeklyAmount",
+            value: "%1$@ weekly",
+            comment: "Total weekly amount of an unlimited Blaze campaign. " +
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift]: Suggestion: I think we can elaborate on the "unlimited" wording. 

```suggestion
            comment: "Total weekly amount of a Blaze campaign configured to run forever with no end date. " +
```

Applies to `totalWeeklyAmountWithCurrency` string as well.

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift [hunk 2]
```diff
@@ -232,6 +245,29 @@ private extension BlazeConfirmPaymentViewModel {
...
+        static let totalWeeklyAmountWithCurrency = NSLocalizedString(
+            "blazeConfirmPaymentViewModel.totalWeeklyAmountWithCurrency",
+            value: "%1$@ %2$@ weekly",
+            comment: "Total weekly amount of an unlimited Blaze campaign. Placeholders are formatted amount and currency. " +
+            "Reads as: $11 USD weekly"
+        )
+        static let totalAmountWithCurrency = NSLocalizedString(
+            "blazeConfirmPaymentViewModel.totalAmountWithCurrency",
+            value: "%1$@ %2$@",
+            comment: "Total amount of an unlimited Blaze campaign. Placeholders are formatted amount and currency. " +
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift]:
```suggestion
            comment: "Total amount of a Blaze campaign. Placeholders are formatted amount and currency. " +
```

**Review by selanthiraiyan — COMMENTED** [review]
The evergreen campaign creation works as described. The code looks good to me. 🚀

**Comment by itsmeichigo** [comment]
Thanks @selanthiraiyan for the reviews! I updated the comments, could you give this PR an approval if this is all good?

**Review by selanthiraiyan — APPROVED** [review]
Thanks for the updates, Huong! The updated strings look good to me. 🚀

**Pull request merged**